lib/internal/fs/promises.js

```diff
@@ -250,6 +251,22 @@ async function open(path, flags, mode) {
 
 async function read(handle, buffer, offset, length, position) {
   validateFileHandle(handle);
+
+  // Arguments are passed from FileHandle.read(),
+  // so arguments.length is always 5,
+  // explicit filtering is needed
+  if (ObjectValues(arguments).filter(
```

Comment:
Perhaps there is a simpler, more efficient check that could be used here instead?

Reply:
This is my first time going inside nodes "guts". I have seen ObjectValues beeing used in other places, so I have decided to use it. If there is a better way to do this, then I'm not aware of it. At least, not in a way that would make this more or equally readable.

Reply:
see https://github.com/nodejs/node/blob/9b8d317d9916dd44b2b3678df05f9daf6eb14c90/lib/fs.js#L476-L496

Reply:
@Himself65 My code is based exactly on fs.read example, you can even see the same format of comment

Reply:
I mean, I prefer to use 
```js
if (arguments.length < 3) {
  // xxx
}
```

`ObjectValues().filter` is redundant.

Reply:
@Himself65 
On lines 255-257:
``` 
// Arguments are passed from FileHandle.read(),
// so arguments.length is always 5,
// explicit filtering is needed
```
Argument length is always 5 regardless of actual arguments passed because of:

https://github.com/nodejs/node/blob/9b8d317d9916dd44b2b3678df05f9daf6eb14c90/lib/internal/fs/promises.js#L109-L111

Arguments that are not passed will be ```undefined```.
For example: arguments of ```filehandler.read(Buffer.alloc(21), 1)``` are ```Arguments(5) [FileHandle, Buffer(21), 1, undefined, undefined]```
You are right about test-case, I will add it today, thanks.

Reply:
Oh, I missed the comment.

You can use

```js
read(...args) { 
   return read(...args); 
 }
```

Reply:
@Himself65 I just tested it, and it works! Major thanks. Will update code and send changes.